packages/background/src/controllers/NetworkController.ts

```diff
@@ -710,23 +713,18 @@ export default class NetworkController extends BaseController<NetworkControllerS
             // check for eip1559 compatibility
             await this.getEIP1559Compatibility(network.chainId, true);
 
-            // Set the isNetworkChanging flag to false
-            this.store.updateState({
-                isNetworkChanging: false,
-            });
-
-            // Emit NETWORK_CHANGE event
-            this.emit(NetworkEvents.NETWORK_CHANGE, this.network);
-
             // Return network change success
             return true;
         } catch (error) {
-            // Set the isNetworkChanging flag to false
-            this.store.updateState({ isNetworkChanging: false });
-
             // If an error was thrown
             // return network change failure
             return false;
+        } finally {
+            // Set the isNetworkChanging flag to false
+            this.store.updateState({ isNetworkChanging: false });
+
+            // Emit NETWORK_CHANGE event
+            this.emit(NetworkEvents.NETWORK_CHANGE, this.network);
```

Comment:
TL;DR
We need to keep every controller synced about the current network. 


The network actually changed. If not, the UI will show the loading while it is changing the network and then it will come back to the old network. 
So, the idea here is to change the network and display the provider down dialog and let the user go back to a healthy network (or change the rpc). We need emit the event so that every controller that listens to the network change is in sync with the new network.

We discovered this by switching from Arbitrum to Mainnet and checking the chrome network tab : there were errors on mainnet's node but chain fee was requesting arbitrum's block number, so that those controllers were out of sync. 


I would also apply another change (in a different Pr) that listen to the provider status events and avoid requesting chain fee if the network provider is down.